src/Pages/Overview/Overview.js

```diff
@@ -34,15 +43,48 @@ export default class OverviewBoard extends Component {
     if (!apiResponse.error) this.setState({ users: apiResponse.responseData })
   }
 
-  updateQuery (event) {
-    let { value } = event.target
+  updateQuery (value) {
+    // taking care of empty values
+    value = typeof value === 'undefined' ? '' : value
     value = value.trim().toLowerCase()
-    this.setState({ query: value })
+    let queryResult
+
+    const userExists = user => {
+      return (
+        user.firstName.toLowerCase().includes(value) ||
+        user.lastName.toLowerCase().includes(value) ||
+        user.email.toLowerCase().includes(value)
+      )
+    }
+
+    if (this.state.currentQueryType === 'All') {
+      queryResult = this.state.users.filter(data => userExists(data))
+    } else if (this.state.currentQueryType === 'Officer') {
+      const filtered = this.state.users.filter(
+        data => data.accessLevel === membershipState.OFFICER
+      )
+      const search = this.state.users.filter(
+        data => userExists(data) && data.accessLevel === membershipState.OFFICER
+      )
+      queryResult = search.length > 0 ? search : filtered
+    } else if (this.state.currentQueryType === 'Admin') {
+      const filtered = this.state.users.filter(
+        data => data.accessLevel === membershipState.ADMIN
+      )
+      const search = this.state.users.filter(
+        data => userExists(data) && data.accessLevel === membershipState.ADMIN
+      )
+      queryResult = search.length > 0 ? search : filtered
+    } else if (this.state.currentQueryType === 'Pending') {
+      const filtered = this.state.users.filter(
```

Comment:
what is the difference between filtered and search? can you combine the two?

Reply:
```queryResult``` holds the final array of users to be map out to the dahsboard.

```filtered``` holds all users with the specified ```queryType of "Officer, Admin, or Pending"``` Render when input is empty or when a different ```queryType``` is selected.

```search``` is ```filtered``` + ```input ``` of "First name, last name, or email" (more narrowed). Render when input is not empty.